crates/flux-fhir-analysis/src/conv/mod.rs

```diff
@@ -1979,6 +1979,10 @@ fn conv_sort_path(
         fhir::SortRes::User { name } => rty::SortCtor::User { name },
         fhir::SortRes::Adt(def_id) => {
             let sort_def = genv.adt_sort_def_of(def_id)?;
+            let generics = genv.generics_of(def_id)?;
+            if !path_generics_correct(path, generics) {
+                return Err(QueryErr::InvalidGenericArg { def_id });
```

Comment:
This function returns a `QueryResult`, but we should emit a specialized error. The trick is to convert a `Result<T, ErrorGuaranteed>` to a `Result<T, QueryErr>` (a.k.a `QueryResult<T>`). See for example how we report [TooFewGenericArgs](https://github.com/flux-rs/flux/blob/e150d6899cd96b9770b5e3ea3a54c0dd78a89be4/crates/flux-fhir-analysis/src/conv/mod.rs#L1458)